bindings/rascal/utils/cg_utils.py

```diff
@@ -0,0 +1,652 @@
+            self._wddict[L] = np.real(c2r_mats[L] @ np.conjugate(wig) @ r2c_mats[L])
+
+    def rotate(self, rho):
+        """
+        Rotates a vector of 2l+1 spherical harmonics (coefficients) according to the
+        rotation defined in the initialization.
+
+        rho: array
+            List of 2l+1 coefficients
+
+        Returns:
+        --------
+        (2l+1) array containing the coefficients for the rotated structure
+        """
+
+        L = (rho.shape[-1] - 1) // 2
+        return rho @ self._wddict[L].T
+
+    def rotate_frame(self, frame, in_place=False):
+        """
+        Utility function to also rotate a structure, given as an Atoms frame.
+        NB: it will rotate positions and cell, and no other array.
+
+        frame: ase.Atoms
+            An atomic structure in ASE format, that will be modified in place
+        in_frame: bool
+            Whether the frame should be copied or processed in place (defaults to False)
+
+        Returns:
+        -------
+        The rotated frame.
+        """
+
+        if is_ase_Atoms(frame):
+            if in_place:
+                frame = frame.copy()
+            frame.positions = frame.positions @ self._rotation.T
+            frame.cell = frame.cell @ self._rotation.T
+        else:
+            if in_place:
+                frame = deepcopy(frame)
+            frame["positions"] = self._rotation @ frame["positions"]
+            frame["cell"] = self._rotation @ frame["cell"]
+        return frame
+
+
+class ClebschGordanReal:
+    """
+    Helper class to manipulate Clebsch-Gordan coefficients with real-storage
+    spherical harmonics and coefficients.
+    """
+
+    def __init__(self, lmax):
+        """
+        Initialize the ClebschGordanReal class, precomputing transformation coefficients
+        up to lmax
+
+        lmax: int
+            Maximum angular momentum channel that is computed
+        """
+
+        self._lmax = lmax
+        self._cgdict = {}
+        self._cgraw = {}
+
+        # real-to-complex and complex-to-real transformations as matrices
+        r2c_mats = {}
+        c2r_mats = {}
+        for L in range(0, self._lmax + 1):
+            r2c_mats[L] = real2complex_matrix(L)
+            c2r_mats[L] = np.conjugate(r2c_mats[L]).T
+
+        for l1 in range(self._lmax + 1):
+            for l2 in range(self._lmax + 1):
+                for L in range(
+                    max(l1, l2) - min(l1, l2), min(self._lmax, (l1 + l2)) + 1
+                ):
+                    # computes CG coefficients
+                    ccg = _cg(l1, l2, L)
+                    self._cgraw[(l1, l2, L)] = ccg
+
+                    # applies transformations that make them act and generate real
+                    # valued coefficients
+                    if (l1 + l2 + L) % 2 == 0:
+                        rcg = np.real(
+                            np.einsum(
+                                "abc, ax, by, zc -> xyz",
```

Comment:
I still this this is worth some explanation for the future readers of this bit of code

Reply:
I added a comment already
```
                    # applies transformations that make them act and generate real
                    # valued coefficients. this is basically achieved taking the
                    # <l1m1; l2m2| LM> "complex" CG and apply a real-to-complex to
                    # the bra (r2c) and a complex-to-real transformation to the ket
                    # (c2r). Note that since the c2r is applied to the right, there
                    # has to be a transposition (zc rather than cz in einsum)
```

Reply:
Ok, I missed this!